applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_2D2N.cpp

```diff
@@ -878,28 +878,28 @@ void CrBeamElement2D2N::AddExplicitContribution(
   }
 
   if (rDestinationVariable == NODAL_INERTIA) {
+        Matrix element_mass_matrix = ZeroMatrix(msElementSize, msElementSize);
+        ProcessInfo temp_info; // Dummy
+        this->CalculateMassMatrix(element_mass_matrix, temp_info);
 
-    Matrix element_mass_matrix = ZeroMatrix(msElementSize, msElementSize);
-    ProcessInfo temp_info; // Dummy
-    this->CalculateMassMatrix(element_mass_matrix, temp_info);
+        for (IndexType i = 0; i < msNumberOfNodes; ++i) {
+            double aux_nodal_mass = 0.0;
+            double aux_nodal_inertia = 0.0;
 
-    for (SizeType i = 0; i < msNumberOfNodes; ++i) {
-      GetGeometry()[i].SetLock();
-      double &r_nodal_mass = GetGeometry()[i].GetValue(NODAL_MASS);
-      array_1d<double, msLocalSize> &r_nodal_inertia =
-          GetGeometry()[i].GetValue(NODAL_INERTIA);
-      SizeType index = i * msLocalSize;
-
-      for (SizeType j = 0; j < msElementSize; ++j) {
-        r_nodal_mass += element_mass_matrix(index, j);
-        r_nodal_inertia[msDimension] +=
-            element_mass_matrix(index + msDimension, j);
-      }
-      for (SizeType k = 0; k < msLocalSize; ++k)
-        r_nodal_inertia[k] = std::abs(r_nodal_inertia[k]);

```

Comment:
here you were doing an assigment, you can not do that. The variable you are computing is changing during the computation (the other threads are doing the same computation), so the you are adding a contaminated absolute value

Reply:
And this is not solved with a critical, trust I tried